### DISCUSSION TITLE
React Native app for internal distribution

### DISCUSSION THREAD

**Discussion by derynL** [codemagic-ci-cd/codemagic-docs#2309] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

No steps

### Expected results

I'm still in discovery mode. My team is building an app that is not for public release. It will be made available to certain employees within the business, so we won't be deploying to App Store. We're using an Apple Developer Enterprise account, which does not provide access to App Store Connect and therefore cannot produce an App Store Connect API key. Is there a way to build our app with codemagic.yaml without the API key? Or should I be looking at alternative pipeline solutions for our particular use case?

### Actual results

N/A

**Comment by icarusdust** [comment, marked as answer]
Hi @derynL, you do not need to provide an API key to build your application. Providing it is one way of doing it, the other way that Codemagic offers to code signing and create your iOS binary is called manual code signing (Code signing identities) with which only a certificate and provisioning profile created by you suffice. Check out [this documentation](https://docs.codemagic.io/yaml-code-signing/signing-ios/) for more information. 


Note: If you are going to test internally, then you will need to choose In House provisioning profile type when creating one in your Apple Developer Enterprise Account.
